.github/workflows/docs_workflow.yml

```diff
@@ -59,14 +59,28 @@ jobs:
           set -e
           pip check
           cd docs
-          sphinx-multiversion . _build/html
+          DOCS_VERSION=${{ github.ref_name }} make versioned-html
       - name: Copy Docs and Commit
         run: |
           set -e
           pip check
           cd docs
           # gh-pages branch must already exist
           git clone https://github.com/MPAS-Dev/geometric_features.git --branch gh-pages --single-branch gh-pages
+
+          # Only replace docs in a directory with the destination branch name with latest changes. Docs for
+          # releases should be untouched.
+          rm -rf gh-pages/${{ github.ref_name }}
+
+          # don't clobber existing release versions (in case we retroactively fixed them)
+          cp -r _build/html/${{ github.ref_name }} gh-pages/
+
+          mkdir -p gh-pages/shared
+          cp shared/version-switcher.js gh-pages/shared/version-switcher.js
+
+          # Update the list of versions with all versions in the gh-pages directory.
+          python generate_versions_json.py
+
```

Comment:
There are also some lines that need to be removed here, see:
https://github.com/E3SM-Project/polaris/commit/3467e5b7dc94da0abdb073283ab358a2d322a1cd